File: internal/yubin/publication.go
package yubin

import "time"

type SendOptions struct {
	Topics   []string
	Users    []string
	SourceID string
	Template string
}

type PublishOptions struct {
	SendOptions SendOptions
	At          *time.Time
	Meta        map[string]any
}

type Publication struct {
	ID   string
	Info PublishOptions
}

File: lib/api/rest/client/js/models/publication.ts
export default class Publication {
    public static fromRaw(data: any) {
        data ||= {}
        return new Publication(
            data["source"] || '',
            data["template"] || '',
            data["users"] || [],
            data['topics'] || [],
            data["meta"] || null,
            data["at"] || 0
        )
    }

    constructor(
        public source: string,
        public template: string,
        public users: string[] | null,
        public topics: string[] | null,
        public meta: any,
        public at: number) {
    }
}